packages/svelte/src/compiler/phases/3-transform/client/visitors/template.js

```diff
@@ -2943,9 +2962,18 @@ export const template_visitors = {
 
 		for (const attribute of node.attributes) {
 			if (attribute.type === 'SpreadAttribute') {
-				spreads.push(
-					b.thunk(/** @type {import('estree').Expression} */ (context.visit(attribute)))
+				const value = /** @type {import('estree').Expression} */ (context.visit(attribute));
+				const keys = b.call(
+					'$.derived',
+					b.thunk(
+						b.call(
+							b.member(b.call('Object.keys', b.logical('??', value, b.object([]))), b.id('join')),
```

Comment:
Why do we join them with commas?

Reply:
Just to avoid possible situations where two consequent keys changes but the resulting join doesnt

```ts
const x = $state({
  foo: true,
  bar: true,
});

x = {
  foob: true,
  ar: true,
}
```

Reply:
Can't we just use the array `['foo', 'bar']` instead of a string?

Reply:
It needs to be a string because it has to have a stable reference